x-pack/plugins/screenshotting/server/screenshots/observable.ts

```diff
@@ -168,52 +170,61 @@ export class ScreenshotObservableHandler {
     const driver = this.driver;
     const waitTimeout = toNumber(this.config.capture.timeouts.waitForElements);
 
-    return defer(() => getNumberOfItems(driver, this.logger, waitTimeout, this.layout)).pipe(
+    return defer(() => getNumberOfItems(driver, this.eventLogger, waitTimeout, this.layout)).pipe(
       mergeMap(async (itemsCount) => {
         // set the viewport to the dimensions from the job, to allow elements to flow into the expected layout
         const viewport = this.layout.getViewport(itemsCount) || getDefaultViewPort();
 
         // Set the viewport allowing time for the browser to handle reflow and redraw
         // before checking for readiness of visualizations.
-        await driver.setViewport(viewport, this.logger);
-        await waitForVisualizations(driver, this.logger, waitTimeout, itemsCount, this.layout);
+        await driver.setViewport(viewport, this.eventLogger.kbnLogger);
+        await waitForVisualizations(driver, this.eventLogger, waitTimeout, itemsCount, this.layout);
       }),
       this.waitUntil(waitTimeout, 'wait for elements')
     );
   }
 
-  private completeRender(apmTrans: Transaction | null) {
+  private completeRender() {
     const driver = this.driver;
     const layout = this.layout;
-    const logger = this.logger;
+    const eventLogger = this.eventLogger;
 
     return defer(async () => {
       // Waiting till _after_ elements have rendered before injecting our CSS
       // allows for them to be displayed properly in many cases
-      await injectCustomCss(driver, logger, layout);
+      await injectCustomCss(driver, eventLogger, layout);
 
-      const apmPositionElements = apmTrans?.startSpan('position-elements', 'correction');
+      this.eventLogger.positionElementsStart();
       // position panel elements for print layout
-      await layout.positionElements?.(driver, logger);
-      apmPositionElements?.end();
+      await layout.positionElements?.(driver, eventLogger.kbnLogger);
+      this.eventLogger.positionElementsEnd();
```

Comment:
Shall we handle the case when "position element" fails?

Reply:
35dd0a80b71